editor/components/block-list/block.js

```diff
@@ -96,10 +102,7 @@ export class BlockListBlock extends Component {
 		// is defined in `@wordpress/blocks`, so to avoid a circular dependency
 		// we inject this function via context.
 		return {
-			createInnerBlockList: ( uid ) => {
```

Comment:
I think it may have been expected that `getChildContext` wasn't called as frequently as it actually is. While this works, an alternative may be to simply update this to the React 16.3.0 syntax. In either case, we should want the function to point to a consistent reference (the legacy API may actually make this easier for object context).

Reply:
Playing with when "context creators" are called: https://codepen.io/aduth/pen/LmZgKN?editors=1111

Reply:
>Anyways, I think that in the long createInnerBlockList could be used directly as a dependency without using context at all.

I'm curious how you'd see this working. Do you mean being explicit with how a block must pass its own `rootUID`? The existence of this context is largely to simplify the implementation behavior for block authors (similar in many ways to the auto-handling of `isSelected`).

Reply:
It should be easier to reason after `Framework: Move components from the blocks to the editor module` (#6521) gets merged. It seems like you would be able to use `createInnerBlockList` as an internal function, but you would still have to pass down `renderBlockMenu`. I would investigate once all `blocks` components are moved to `editor`.